daemon/cluster/listen_addr.go

```diff
@@ -123,6 +123,22 @@ func validateDefaultAddrPool(defaultAddrPool []string, size uint32) error {
 	return nil
 }
 
+// validateDataPathPort validates vxlan udp port (data path port) number.
+func validateDataPathPort(portNum uint32) error {
+	// if the value comes as 0 by any reason we set it to default value 4789
+	if portNum == 0 {
+		portNum = 4789
```

Comment:
So; does this actually work? Here's a quick test; (you can run it here: https://play.golang.org/p/MnC5gxYX26F)

```go
package main

import (
	"fmt"
)

type InitRequest struct {
	DataPathPort uint32
}

func main() {
	req := InitRequest{}
	if err := validateDataPathPort(req.DataPathPort); err != nil {
		fmt.Println(err)
	}
	fmt.Printf("Data port number: %d", req.DataPathPort)
}

func validateDataPathPort(portNum uint32) error {
	// if the value comes as 0 by any reason we set it to default value 4789
	if portNum == 0 {
		portNum = 4789
	}
	// IANA procedures for each range in detail
	// The Well Known Ports, aka the System Ports, from 0-1023
	// The Registered Ports, aka the User Ports, from 1024-49151
	// The Dynamic Ports, aka the Private Ports, from 49152-65535
	// So we can allow range between 1024 to 49151
	if portNum < 1024 || portNum > 49151 {
		return fmt.Errorf("validateDataPathPort : Data Path Port number is not in valid range %d", portNum)
	}
	return nil
}
```

I'm not always a big fan of validation-functions modifying the value; I see we do this in other places, but perhaps for this one we should change the signature to something like;

```go
func getDataPathPort(portNum uint32) (uint32, error) {
    // ....
}
```

Or pass the whole `req` (as is done in `validateAndSanitizeInitRequest`);

```go
func validateDataPathPort(req *types.InitRequest) error {
    // ....
}
```

Reply:
Actually; looks like we're still setting the default in libnetwork as well. Would it be better to just pass `0`, and let libnetwork set the default?

Reply:
by looking at other validation/resolve/get APIs, I feel we can stick to your below suggestion. I will change it accordingly.
func getDataPathPort(portNum uint32) (uint32, error) {
    // ....
}

Reply:
> Actually; looks like we're still setting the default in libnetwork as well. Would it be better to just pass `0`, and let libnetwork set the default?

This value gets stored into swarm too. So I better change it like I proposed in the above reply.